tests/test_comparison.py

```diff
@@ -3,86 +3,98 @@
-    with contextlib.redirect_stdout(io.StringIO()) as buf:
-        task.run()
-    assert (
-        buf.getvalue() == f"Success: files {file_a.name} {file_b.name} are identical\n"
+    _comparison_task = ComparisonTask(
+        files=files, task_name="mock_comparison_task_name"
     )
+    _comparison_task.subprocess_handler = mock_subprocess_handler
+    _comparison_task.root_dir = mock_cwd
+    return _comparison_task
 
-    # Success case: test verbose output
-    task = get_mock_comparison_task()
-    with contextlib.redirect_stdout(io.StringIO()) as buf:
-        task.run(verbose=True)
-    assert buf.getvalue() == (
-        f"Comparing files {file_a.name} and {file_b.name} bitwise...\n"
-        f"Success: files {file_a.name} {file_b.name} are identical\n"
-    )
 
-    # Failure case: test failed comparison check (files differ)
-    mock_subprocess = MockSubprocessWrapper()
-    mock_subprocess.error_on_call = True
-    task = get_mock_comparison_task(subprocess_handler=mock_subprocess)
-    stdout_file = bitwise_cmp_dir / f"{task.task_name}.txt"
-    task.run()
-    with open(stdout_file, "r", encoding="utf-8") as file:
-        assert file.read() == mock_subprocess.stdout
-
-    # Failure case: test non-verbose standard output on failure
-    mock_subprocess = MockSubprocessWrapper()
-    mock_subprocess.error_on_call = True
-    task = get_mock_comparison_task(subprocess_handler=mock_subprocess)
-    stdout_file = bitwise_cmp_dir / f"{task.task_name}.txt"
-    with contextlib.redirect_stdout(io.StringIO()) as buf:
-        task.run()
-    assert buf.getvalue() == (
-        f"Failure: files {file_a.name} {file_b.name} differ. Results of diff "
-        f"have been written to {stdout_file}\n"
-    )
+class TestRun:
+    """Tests for `ComparisonTask.run()`."""
 
-    # Failure case: test verbose standard output on failure
-    mock_subprocess = MockSubprocessWrapper()
-    mock_subprocess.error_on_call = True
-    task = get_mock_comparison_task(subprocess_handler=mock_subprocess)
-    stdout_file = bitwise_cmp_dir / f"{task.task_name}.txt"
-    with contextlib.redirect_stdout(io.StringIO()) as buf:
-        task.run(verbose=True)
-    assert buf.getvalue() == (
-        f"Comparing files {file_a.name} and {file_b.name} bitwise...\n"
-        f"Failure: files {file_a.name} {file_b.name} differ. Results of diff "
-        f"have been written to {stdout_file}\n"
-    )
+    @pytest.fixture
+    def bitwise_cmp_dir(self, mock_cwd):
+        _bitwise_cmp_dir = mock_cwd / internal.FLUXSITE_BITWISE_CMP_DIR
+        _bitwise_cmp_dir.mkdir(parents=True)
+        return _bitwise_cmp_dir
+
+    def test_nccmp_execution(self, comparison_task, files, mock_subprocess_handler):
+        """Success case: test nccmp is executed"""
+        file_a, file_b = files
+        comparison_task.run()
+        assert f"nccmp -df {file_a} {file_b}" in mock_subprocess_handler.commands
+
+    def test_default_standard_output(self, comparison_task, files):
```

Comment:
I wonder if we could instead use the test [parameterization](https://docs.pytest.org/en/7.1.x/how-to/parametrize.html) to write one output test that runs on all output cases. But this might wait for after the reorganisation?

Reply:
It should be possible to use parameterisation for testing different verbosity levels. I'll look into it